src/library/Table/TableSortableHeaderCell.js

```diff
@@ -0,0 +1,198 @@
+        outline: 0
+      },
+
+      '&::-moz-focus-inner': {
+        border: 0
+      },
+
+      // [1]
+      '&::after': {
+        bottom: '100%',
+        content: '""',
+        height: '10em',
+        left: 0,
+        position: 'absolute',
+        right: 0
+      }
+    };
+  },
+  content: {
+    whiteSpace: 'normal'
+  },
+  iconHolder: ({ isActiveSort, direction, theme }) => {
+    const iconAdjustment = pxToEm(2);
+    const space = `${parseFloat(theme.space_inline_xxs) +
+      parseFloat(iconAdjustment)}em`;
+
+    return {
+      color: theme.icon_color,
+      display: 'inline-block',
+      height: '0.875em',
+      marginLeft: theme.direction === 'ltr' ? space : null,
+      marginRight: theme.direction === 'rtl' ? space : null,
+      opacity: isActiveSort ? null : 0,
+      position: 'relative',
+      top: direction === 'ascending' ? 2 : 1,
+      width: '0.875em',
+
+      '& > [role="img"]': {
+        margin: `-${iconAdjustment}`
+      },
+
+      '*:hover > button > &, button:focus > &': {
+        color: 'inherit',
+        opacity: 1
+      }
+    };
+  }
+};
+
+const Root = createStyledComponent(TableHeaderCell, styles.root, {
+  displayName: 'TableSortableHeaderCell',
+  withProps: { noPadding: true }
+});
+const Button = createStyledComponent(TableHeaderCell, styles.button, {
+  withProps: { element: 'button' }
+});
+const Content = createStyledComponent('span', styles.content);
+const IconHolder = createStyledComponent('span', styles.iconHolder);
+
+const iconProps = {
+  'aria-hidden': true,
+  size: 'auto'
+};
+const sortIcon = {
+  ascending: <IconArrowDropdownUp {...iconProps} />,
+  descending: <IconArrowDropdownDown {...iconProps} />
+};
+
+export default function TableSortableHeaderCell({
+  children,
+  label,
+  name,
+  messages,
+  sortable,
+  ...restProps
+}: Props) {
+  const { sort, sortFn } = sortable;
+
+  const sortColumn = sort && sort.key;
+  const descending = sort ? sort.descending : false;
+
+  const isActiveSort = sortColumn === name;
+  const activeDirection = descending ? 'descending' : 'ascending';
+  const nextDirection =
+    isActiveSort && activeDirection === 'ascending'
+      ? 'descending'
+      : 'ascending';
+
+  const toggle = isActiveSort || activeDirection !== nextDirection;
+  const nextDescending = toggle ? !descending : descending;
+
+  const a11yLabel = label || children;
+
+  const rootProps = {
+    ...restProps,
+    // TODO: Mac Chrome VO (others?) announces this twice?
+    'aria-label': a11yLabel,
+    'aria-sort': isActiveSort ? activeDirection : 'none',
+    role: 'columnheader'
+  };
+
+  const buttonProps = {
+    ...restProps,
+    'aria-label':
+      nextDirection === 'ascending'
+        ? messages.sortColumnAscending
+        : messages.sortColumnDescending,
+    onClick: () => {
+      sortFn({ key: name, descending: nextDescending });
+    }
+  };
+  const iconHolderProps = {
+    direction: (isActiveSort && activeDirection) || 'ascending',
```

Comment:
Something about the lack of icon change when I click a column to sort feels wrong to me. Instead of 'ascending' as our default here, could we have an icon with both arrows?

Reply:
For now, I removed the arrow on hover. We'll file a follow-up story to take a better look at this.